src/use_cases/ResetBalance/ResetBalanceInteractor.java

```diff
@@ -0,0 +1,49 @@
+package use_cases.ResetBalance;
+
+import entities.*;
+import interface_adapters.ResetBalance.ResetBalanceOutputBoundary;
+import use_cases.APIAccessInterface;
+import use_cases.BaseStockInteractor;
+
+import java.time.LocalDate;
+import java.util.HashMap;
+
+public class ResetBalanceInteractor extends BaseStockInteractor implements ResetBalanceInputBoundary {
+    final ResetBalanceDataAccessInterface userDataAccessObject;
+    ResetBalanceOutputBoundary resetBalancePresenter;
+    APIAccessInterface driverAPI;
+
+    public ResetBalanceInteractor(ResetBalanceDataAccessInterface userDataAccessInterface,
+                                  ResetBalanceOutputBoundary resetBalancePresenter, APIAccessInterface driverAPI) {
+        super(driverAPI);
+        this.userDataAccessObject = userDataAccessInterface;
+        this.resetBalancePresenter = resetBalancePresenter;
+        this.driverAPI = driverAPI;
+    }
+    @Override
+    public void execute(ResetBalanceInputData sellInputData) {
+        String username = sellInputData.getUsername();
+
+        User user = userDataAccessObject.get(username);
+
+        if (user.getBalance() >= 500.0) {
+            resetBalancePresenter.prepareFailView("You have way too much money.");
+            return;
+        }
+
+        Double curBalance = user.getBalance();
+        Double amountToAdd = 10000.0;
+        user.addBalance(amountToAdd);
+
+        // honestly, maybe consider adding a TopupTransaction
+        HashMap<String, TransactionHistory> userHistory = user.getHistory();
+
+        Transaction transaction = new TopupTransaction(curBalance + amountToAdd, new PricePoint(LocalDate.now(), curBalance));
+
+        super.addToHistory(userHistory, "Stonks App", user, curBalance + amountToAdd, curBalance, transaction);
```

Comment:
This may come up as a major issue in the future. How do you plan on mitigating that "Stonks App" is in place of the stock ticker parameter?

Reply:
"Stonks App" is indeed not a stock ticker, however we would treat this as a special key which would be handled from all the other keys. The reason as to why it isn't a ticker is so that we could differentiate between profits and balance resets in the History. Think of this as a special case.